AWGL/somatic_db
# Home page

## Change description
- Update static files for bootstrap, font awesome and jquery
- Add a home page - also includes three AJAX calls for interactive bits - search bar and status boxes
- Added upload date to worksheets model and import management command
- Updated bootstrap breadcrumbs because of new bootstrap version
- View samples page can now show either all samples assigned to a user, or all samples on a worksheet, they use the same view but different URLs, controlled by optional kwargs
- Moved some JS and CSS into static folders rather than within HTML template. Rest will be done in future commit


> **Looking at the diff**
> Use this link to compare code, otherwise all static will be included and it'll probably crash your browser - https://github.com/AWGL/somatic_db/compare/development..home_page (important that it has two dots and not three, three will still include all the static)
> Ignore changes to:
>   - analysis/templates/analysis/analysis-snvs.html
>   - analysis/models.py - Gnomad stuff, line 264 onwards
>   - analysis/test_data/Database_37/crm_test_1_variants.tsv
>   - analysis/tests.py
>   - analysis/utils.py - Gnomad stuff at end

## Justification for minor update
No effect on processing of data, only how it's visualised

## Testing description
- Running existing unit tests
- Check that static is loaded in correctly (no loading errors in console, bootstrap renders nicely, datatables still renders etc)
- Make sure breadcrumbs are rendered nicely (not bullet points)
- Check search function on home page works for worksheet, sample and run ID, should take to worksheets page on click
- Check the counts of 'samples assigned to me' and 'worksheets pending' are correct and that the links take you to the samples/worksheets in question
- Check the colour of the two above is green when zero and orange when not
- Check that the quick search links work

## Test data location
- Existing test data, some samples need to be assigned/unassigned/ completed etc to fully test all the functionality as described above

## Verification results
- All unit tests pass
- All behaved as expected
- Deployment - will need migrations but its on an optional field so should be straightforward
